tf/components/kubernetes.tf

```diff
@@ -1,3 +1,153 @@
 provider "kubernetes" {
   version = "~> 1.12"
+}
+
+resource "kubernetes_namespace" "default_namespace" {
+  metadata {
+    name = var.namespaces
+  }
+
+  depends_on = [
+    helm_release.argocd_release,
+  ]
+}
+
+# resource "kubernetes_secret" "docker_pull_secret" {
+#   metadata {
+#     name = "docker-registry"
+#     namespace = kubernetes_namespace.default_namespace.metadata.0.name
+#   }
+
+#   data = {
+#     ".dockerconfigjson" = "${file("../env/docker-registry.json")}"
+#   }
+
+#   type = "kubernetes.io/dockerconfigjson"
+# }
+
+resource "null_resource" "secret" {
+  provisioner "local-exec" {
+    command = "kubectl expose deployment argocd-server -n argocd --type=NodePort --name argocd-server-exposed"
+  }
+  provisioner "local-exec" {
+    command = "kubectl patch svc argocd-server-exposed -n argocd --type=json -p '[{\"op\": \"replace\", \"path\": \"/spec/type\", \"value\": \"NodePort\"}, {\"op\": \"replace\", \"path\": \"/spec/ports/0/nodePort\", \"value\" : 30080}]'"
+    interpreter = ["PowerShell", "-Command"]
+  }
+  # This step will FAIL unless port-forwarding is on, WIP
+  provisioner "local-exec" {
+    command = "argocd login ${minikube_ip}:30080 --username admin --password ${random_password.argopass.result} --insecure | argocd repo add https://github.com/${var.git_server}/${var.git_repository} --username ${var.git_username} --password ${var.git_password} --name ${var.git_repository} "
+    interpreter = ["PowerShell", "-Command"]
```

Comment:
This one is not compatible with mac. Should work for both mac and window?

Reply:
```suggestion
```

Reply:
Can be skipped in most of the commands, however there is an issue in the patch command since there is a bug in the default windows cmd interpreter where the escape character doesn't disappear on execution, thus leading to an error. Any known workarounds on this?

Reply:
Ok, I will merge to `master`